pkg/sql/codegen/couler/katib.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couler
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/sql/codegen/attribute"
+	"sqlflow.org/sqlflow/pkg/sql/ir"
+)
+
+var defaultKatibDockerImage = "sqlflow/sqlflow"
+
+var attributeDictionary = attribute.Dictionary{
+	"eta": {attribute.Float, `[default=0.3, alias: learning_rate]
+Step size shrinkage used in update to prevents overfitting. After each boosting step, we can directly get the weights of new features, and eta shrinks the feature weights to make the boosting process more conservative.
+range: [0,1]`, attribute.Float32RangeChecker(0, 1, true, true)},
+	"num_class": {attribute.Int, `Number of classes.
+range: [2, Infinity]`, attribute.IntLowerBoundChecker(2, true)},
+	"objective":       {attribute.String, `Learning objective`, nil},
+	"range.num_round": {attribute.IntList, `[ default=[50, 100] ] The range of number of rounds for boosting.`, nil},
+	"range.max_depth": {attribute.IntList, `[ default=[2, 8] ] The range of max depth during training.`, nil},
+	"validation.select": {attribute.String, `[default=""]
+Specify the dataset for validation.
+example: "SELECT * FROM boston.train LIMIT 8"`, nil},
+}
+
+func resolveModelType(estimator string) (string, string, error) {
```

Comment:
How about using  the existing code to reduce the code: https://github.com/sql-machine-learning/sqlflow/blob/0d60f92538c896e61ef869b98773540a9e019c1f/pkg/sql/codegen/xgboost/codegen.go#L47-L56

Reply:
They are different. My function returns 3 values.